src/theory/quantifiers/extended_rewrite.cpp

```diff
@@ -718,6 +725,84 @@ Node ExtendedRewriter::extendedRewriteBcp(
   return Node::null();
 }
 
+Node ExtendedRewriter::extendedRewriteEqRes(Kind andk,
+                                            Kind ork,
+                                            Kind eqk,
+                                            Kind notk,
+                                            std::map<Kind, bool>& bcp_kinds,
+                                            Node n,
+                                            bool isXor)
+{
+  Assert(n.getKind() == andk || n.getKind() == ork);
+  Trace("ext-rew-eqres") << "Eq res: **** INPUT: " << n << std::endl;
+
+  NodeManager* nm = NodeManager::currentNM();
+  Kind nk = n.getKind();
+  bool gpol = (nk == andk);
+  for (unsigned i = 0, nchild = n.getNumChildren(); i < nchild; i++)
+  {
+    Node lit = n[i];
+    if (lit.getKind() == eqk)
+    {
+      Node eq;
+      if (gpol == isXor)
```

Comment:
I don't quite understand why it is ok to turn `(= a b) V ...` into `(= (not a) b) V ...` (I understand the case where you have `(xor a b) /\ ...` and turn it into `(= (not a) b)`).

Reply:
This isn't turning `(= a b) V ...` into `(= (not a) b) V ...`.
Instead, "eq" here is the equality that can be used as a substitution.  So for example, for `(not a) = b V ( b ^ a )`, this can be seen as `a = b => ( b ^ a )`, which we can use substitution to get `(not a) = b V ( b ^ a ){ a -> b }` which simplifies `(not a) = b V b`.  (which then by BCP simplifies to `a V b`).

Reply:
That makes sense.